tools/HelpGeneration/HelpGeneration.psm1

```diff
@@ -0,0 +1,343 @@
+                            {
+                                break
+                            }
+                        }
+
+                        if (!($foundDescription))
+                        {
+                            $fileErrors += "No description found"
+                        }
+                    }
+                    "## EXAMPLES"
+                    {
+                        # Move the index to the start of the PowerShell code
+                        while ($content[$idx] -notcontains "``````")
+                        {
+                            $idx++
+                        }
+
+                        # Check for the platyPS example template
+                        # 
+                        # ```
+                        # PS C:\> {{ Add example code here }}
+                        # ```
+                        # 
+                        if ($content[$idx+1] -contains "PS C:\> {{ Add example code here }}")
+                        {
+                            $fileErrors += "No examples found"
+                        }
+
+                        # Check for other missing example formats (such as empty)
+                        # 
+                        # ```
+                        # 
+                        # ```
+                        # 
+                        if ([string]::IsNullOrWhiteSpace("$($content[$idx+1])"))
+                        {
+                            $fileErrors += "No examples found"
+                        }
+                    }
+                    "*@{Text=}*"
+                    {
+                        # This case occurs when there is no description provided for a parameter
+                        $parameter = $content[$idx-1].Substring(5)
+                        $fileErrors += "No description found for parameter $parameter"
+                    }
+                    default
+                    {
+
+                    }
+                }
+            }
+            
+            # If the markdown file had any missing help, add them to the list to be printed later
+            if ($fileErrors.Count -gt 0)
+            {
+                $fileExceptions = $Exceptions | where { $_.Target -eq "$file" }
+                
+                $fileErrors | foreach {
+                    $error = $_
+
+                    if (($fileExceptions | where { $_.Description -eq "$error" }) -ne $null)
+                    {
+                        # "Caught error - $($HelpFolder.Parent.Name),$file,$error"                        
+                    }
+                    else
+                    {
+                        $errors += "$($HelpFolder.Parent.Name),$file,$error"
+                    }
+                }
+            }
+        }
+
+        # If there were any errors recorded, print them out and throw
+        if ($errors.Count -gt 0)
+        {
+            $errors | foreach { Add-Content "$PSScriptRoot\..\..\src\Package\ValidateHelpExceptions.csv" $_ }
```

Comment:
I think this outupe file should be a parameter.  Also, I think you want this under ```src\Package\{buildconfig}``` by default in order to get it included in the build.  Finally, this should be called ValdateHelpIssues.csv to be consistent with the other products of static analysis.